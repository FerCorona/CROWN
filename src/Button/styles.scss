.CROWNButton {
    font-family: -apple-system, BlinkMacSystemFont, sans-serif;
    text-align: center;
    display: inline-block;
    font-size: 13px;
    font-weight: 600;
    width: -webkit-fill-available;
    margin-top: 5px;
    margin-bottom: 5px;
    padding: 15px 30px;
    cursor: pointer;
    &.Round{
      border-radius: 30px;
    }
    &.SemiRound{
      border-radius: 10px;
    }
}
.CROWNButton:hover{
  opacity: 0.9;
}