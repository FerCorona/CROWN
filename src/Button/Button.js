import React, { Component } from 'react';

require('./styles.scss');

class Button extends Component {
  constructor() {
    super();
    this.state = { };
  }

  render() {
    const { text, onClick, color, textColor, borderColor, type, shape } = this.props;
    const style = {};
    switch (type) {
      case 'Secondary':
        style.border = ` 2px solid var(${borderColor})`;
        style.color = `var(${textColor})`;
        break;
      case 'Text':
        style.border = 'none';
        style.color = `var(${textColor})`;
        style.boxShadow = 'none';
        break;
      default:
        style.backgroundColor = `var(${color})`;
        style.color = `var(${textColor})`;
        break;
    };
    return (
      <a
        className={`CROWNButton ${shape}`}
        style={style}
        onClick={onClick}
      >
        {text}
      </a>
    );
  }
}

Button.defaultProps = {
  text: 'Click Me!',
  onClick: () => console.log('You clicked on me!'),
  color: '--blue-4',
  textColor: '--white',
  borderColor: '--blue-4',
  type: 'Default',
  shape: 'Round'
};

export default Button;