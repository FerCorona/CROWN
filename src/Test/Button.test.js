import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Button />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});