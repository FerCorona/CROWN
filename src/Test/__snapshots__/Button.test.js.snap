// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="CROWNButton Round "
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "var(--blue-4)",
      "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
      "color": "var(--white)",
    }
  }
>
  Click Me!
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="CROWNButton Round "
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "var(--blue-4)",
      "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
      "color": "var(--white)",
    }
  }
>
  Click Me!
</a>
`;
